src/ansys/mapdl/core/mapdl_geometry.py

```diff
@@ -120,62 +157,428 @@ def _load_iges(self):
             )
         return Iges(self._mapdl._generate_iges())
 
-    def _reset_cache(self):
+    def _reset_cache(self) -> None:
         self._keypoints_cache = None
         self._lines_cache = None
 
+    def __setitem__(self, key: Any, value: Any):
+        raise NotImplementedError("This method has not been implemented yet")
+
+    def __getitem__(self, name: str):
+        name = name.lower()
+        if "kp" in name:
+            return self.keypoints[name]
+        elif "line" in name:
+            return self.lines[name]
+        elif "area" in name:
+            return self.areas[name]
+        elif "volume" in name:
+            return self.volumes[name]
+        else:
+            raise ValueError(
+                f"The entity named '{name}' does not exist as a geometry entity."
+            )
+
     @property
-    def _keypoints(self):
+    def _keypoints(self) -> Tuple[NDArray, NDArray]:
         """Returns keypoints cache"""
         if self._keypoints_cache is None:
             self._keypoints_cache = self._load_keypoints()
         return self._keypoints_cache
 
     @property
-    def keypoints(self):
-        """Keypoint coordinates"""
-        return np.asarray(self._keypoints.points)
+    def keypoints(self) -> pv.MultiBlock:
+        """Obtain the keypoints geometry
+
+        Obtain the selected keypoints as a :class:`pyvista.MultiBlock` object.
+
+        Returns
+        -------
+        pv.MultiBlock
+
+        Examples
+        --------
+
+        >>> mapdl.geometry.keypoints
+        MultiBlock (0x147a78220)
+          N Blocks    26
+          X Bounds    -0.016, 0.016
+          Y Bounds    -0.008, 0.018
+          Z Bounds    -0.003, 0.015
+
+        You can plot the entities:
+
+        >>> mapdl.geometry.keypoints.plot()
+
+        You can access the individual keypoints using indexing:
+
+        >>> keypoint0 = mapdl.geometry.keypoints[0]
+        >>> keypoint0
+        pyvista_ndarray([ 0.     ,  0.01778, -0.00318])
+
+        Or using the entity name:
+
+        >>> kp1 = mapdl.geometry.keypoints["kp 1"]
+        >>> kp1
+        pyvista_ndarray([ 0.     ,  0.01778, -0.00318])
+
+        You can use this to iterate over the different elements:
```

Comment:
What is "this"? A pronoun here can lead to confusion. Are you refering to the preceding code? If so, write "You can use the preceding code to iterate...."